packages/vitest/src/node/pool.ts

```diff
@@ -91,10 +94,18 @@ export function createWorkerPool(ctx: Vitest): WorkerPool {
           id: ++id,
         }
 
-        await pool.run(data, { transferList: [workerPort], name })
-        port.close()
-        workerPort.close()
+        try {
+          await pool.run(data, { transferList: [workerPort], name })
+        } finally {
+          port.close()
+          workerPort.close()
+        }
       }))
+
+      const errors = results.filter((r): r is PromiseRejectedResult => r.status === 'rejected').map(r => r.reason)
+      if (errors.length > 0) {
+        throw new AggregateError(errors)
```

Comment:
Maybe we can add a message pointing to read serialized error? Like, `Errors occurred while running tests. For more information, see serialized error" (or maybe something more specific)